ionoscloud/resource_dataplatform_cluster.go

```diff
@@ -27,16 +27,17 @@ func resourceDataplatformCluster() *schema.Resource {
 			"datacenter_id": {
 				Type:        schema.TypeString,
 				Description: "The UUID of the virtual data center (VDC) in which the cluster is provisioned",
-				ValidateDiagFunc: validation.ToDiagFunc(validation.All(validation.StringLenBetween(32, 63),
-					validation.StringMatch(regexp.MustCompile("^[0-9a-f]{8}-?[0-9a-f]{4}-?[0-9a-f]{4}-?[0-9a-f]{4}-?[0-9a-f]{12}$"), ""))),
+				ValidateDiagFunc: validation.AllDiag(validation.ToDiagFunc(validation.StringLenBetween(32, 63)),
+					validation.ToDiagFunc(validation.StringMatch(regexp.MustCompile("^[0-9a-f]{8}-?[0-9a-f]{4}-?[0-9a-f]{4}-?[0-9a-f]{4}-?[0-9a-f]{12}$"), ""))),
 				Required: true,
 				ForceNew: true,
 			},
 			"name": {
-				Type:             schema.TypeString,
-				Description:      "The name of your cluster. Must be 63 characters or less and must be empty or begin and end with an alphanumeric character ([a-z0-9A-Z]). It can contain dashes (-), underscores (_), dots (.), and alphanumerics in-between.",
-				ValidateDiagFunc: validation.ToDiagFunc(validation.All(validation.StringLenBetween(0, 63), validation.StringMatch(regexp.MustCompile("^[A-Za-z0-9][-A-Za-z0-9_.]*[A-Za-z0-9]$"), ""))),
-				Required:         true,
+				Type:        schema.TypeString,
+				Description: "The name of your cluster. Must be 63 characters or less and must be empty or begin and end with an alphanumeric character ([a-z0-9A-Z]). It can contain dashes (-), underscores (_), dots (.), and alphanumerics in-between.",
+				ValidateDiagFunc: validation.AllDiag(validation.ToDiagFunc(validation.StringLenBetween(0, 63)),
```

Comment:
Could make a custom validation `VerifyAlphanumericString` or similar which also takes a length parameter and maybe a separator slice for the `_ - .` to merge these two into. 
This is already used in 2 different places, and might be used even more in the future.

Reply:
good idea, but not necessary for this case